src/TinyCiv.Server/Handlers/UnitMoveHandler.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.AspNetCore.SignalR;
+using TinyCiv.Server.Core.Extensions;
+using TinyCiv.Server.Core.Services;
+using TinyCiv.Shared;
+using TinyCiv.Shared.Events.Client;
+using TinyCiv.Shared.Events.Server;
+using TinyCiv.Shared.Game;
+
+namespace TinyCiv.Server.Handlers
+{
+    public class UnitMoveHandler : ClientHandler<MoveUnitClientEvent>
+    {
+        private readonly IMapService _mapService;
+
+        private readonly object _movementLocker = new();
+
+        public UnitMoveHandler(IMapService mapService)
+        {
+            _mapService = mapService;
+        }
+
+        protected override async Task OnHandleAsync(IClientProxy caller, IClientProxy all, MoveUnitClientEvent @event)
+        {
+            var unitMovementTuple = (@event.UnitId, new ServerPosition { X = @event.X, Y = @event.Y });
+
+            lock (_movementLocker)
+            {
+                // Yes, it is being checked 2 times :| Just put the validation into another function?
+                if (IgnoreWhen(@event))
+                {
+                    return;
+                }
+
```

Comment:
Hmm, probably need to rethink ```IgnoreWhen()``` design, did not consider that it would need these things

Basically, I wanted it to handle events that are not applicable for the current game state e.g. if we are in the lobby and server sends related with moving - ignore

Probably could use ```IgnoreWhen()``` for that purpose and extract this game validation logic to another function?